### PR TITLE
Convert data dumps page to stimulus.js

### PR DESCRIPTION
**Pull request by martinemde**
I'm targeting the static pages for the first to receive the new design.

The data dump page uses javascript to fill the available dumps. Without a template based approach, it's difficult to switch to Tailwind. Rather than fuss with the original javascript, I quickly rewrote the page using stimulus controllers. It otherwise looks identical.

### CODE REVIEW

File: app/views/pages/data.html.erb [hunk 1]
```diff
@@ -3,8 +3,16 @@
...
   <p>The <a href="https://github.com/rubygems/rubygems.org/tree/master/script/load-pg-dump">load-pg-dump</a> script is
-     available as an example of how to to download and load the most recent weekly dump.</p>
-  <ul class="rubygems-dump-listing-postgresql"></ul>
+    available as an example of how to to download and load the most recent weekly dump.</p>
+  <ul data-controller="dump" data-dump-type-value="postgresql">
+    <template data-dump-target="template">
+      <li><a href="#"></a></li>
+    </template>
+  </ul>
   <p>We also provide weekly dumps of the historial RubyGems.org Redis data. (We do not use redis anymore but these are here for historical purposes.)</p>
```
Comment (martinemde) [inline app/views/pages/data.html.erb:12]: The latest redis dump is 2016. Can we drop it from the page?

Comment (segiddins) [reply app/views/pages/data.html.erb:12]: IMO yes.

File: app/javascript/controllers/dump_controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        Size: item.getElementsByTagName('Size')[0].textContent,
+        StorageClass: item.getElementsByTagName('StorageClass')[0].textContent
+      };
+    });
+  }
+
+  render(files) {
+    files
+      .sort((a, b) => Date.parse(b.LastModified) - Date.parse(a.LastModified))
+      .filter(item => 'STANDARD' === item.StorageClass)
```
Comment (segiddins) [inline app/javascript/controllers/dump_controller.js:40]: Nit: filter before sort is faster

**Review by segiddins — APPROVED** [review]

File: app/javascript/controllers/dump_controller.js (hunk 2 above)
Comment (martinemde) [reply app/javascript/controllers/dump_controller.js:40]: Good catch. I didn't do much more than copy the original javascript (removing jquery).

File: app/views/pages/data.html.erb (hunk 1 above)
Comment (martinemde) [reply app/views/pages/data.html.erb]: I'll remove it. I can't see a reason why we'd need 8 year old redis backups.

**Pull request merged**